tests/test_fmi3.py

```diff
@@ -81,6 +90,49 @@ def test_load_kind_SE(self, ref_fmu):
         with pytest.raises(FMUException, match = re.escape(msg)):
             load_fmu(ref_fmu, kind = "SE")
 
+    def test_get_model_identifier(self):
+        """Test that model identifier is retrieved as expected."""
+        fmu = load_fmu(FMI3_REF_FMU_PATH / "VanDerPol.fmu") # any FMI3 ME would suffice
+        assert fmu.get_identifier() == 'VanDerPol'
+
+    def test_get_get_version(self):
+        """Test that FMI version is retrieved as expected."""
+        fmu = load_fmu(FMI3_REF_FMU_PATH / "VanDerPol.fmu") # any FMI3 ME would suffice
+        assert fmu.get_version() == '3.0'
+
+    def test_instantiation(self, tmpdir):
+        """ Test that instantion works by verifying the output in the log.
+        """
+        found_substring = False
+        substring_to_find = 'Successfully loaded all the interface functions'
+
+        with temp_dir_context(tmpdir) as temp_path:
+             # any FMI3 ME would suffice, log_level set to 5 required by test
+            fmu = load_fmu(FMI3_REF_FMU_PATH / "VanDerPol.fmu", log_level=5)
+            with open(fmu.get_log_filename(), 'r') as f:
+                contents = f.readlines()
+
+        for line in contents:
+            if substring_to_find in line:
+                found_substring = True
+                break
+
+        log_file = ''.join(contents)
+        assert found_substring, f"Unable to locate substring '{substring_to_find}' in file with contents '{log_file}'"
+
+    @pytest.mark.parametrize("ref_fmu", [
+        FMI3_REF_FMU_PATH / "BouncingBall.fmu",
+        FMI3_REF_FMU_PATH / "Dahlquist.fmu",
+        FMI3_REF_FMU_PATH / "Resource.fmu",
+        FMI3_REF_FMU_PATH / "StateSpace.fmu",
+        FMI3_REF_FMU_PATH / "Feedthrough.fmu",
+        FMI3_REF_FMU_PATH / "Stair.fmu",
+        FMI3_REF_FMU_PATH / "VanDerPol.fmu",
+    ])
+    def test_load_kind_auto(self, ref_fmu):
+        """Test initialize all the ME reference FMUs. """
+        fmu = load_fmu(ref_fmu)
+        fmu.initialize() # Should simply pass without any exceptions
 
```

Comment:
I propose to add a few more unit tests for the various functions added:

1. get_default_experiment_(start|stop|tolerace) (possibly in test_fmi.py & parameterize for FMI1,2,3)
2. manually calling enter/exit init mode
3. enter_continuous_time_mode
4. enter_event_mode

Reply:
I added test for this in 2b3b66e but only for FMI 3. I don't think we should over-test these things right now as we do have tests for it but in other places.
Also I cannot add tests yet for `enter_continuous_time_mode` nor `enter_event_mode` since it is part of the next effort for the simulation (I currently get an error invalid sequence of function calls).

Reply:
What is wrong with entering contnuous time more/event mode? Unless there was some change for the FMI3 state machine I missed, these should simply work when done after initialization.

Reply:
I must have done something wrong when I first tested it, I tried just now again and it works as expected, see 79f032b.

Reply:
Question, (not sure if we should have unit tests for this):
How does it look like for invalid call sequences? 

I recall that a double `terminate()` would simply leave an extra "error log message", but nothing else on the Python side. Which is fine for `terminate()`, but just wondering how it looks like with the rest.